Shape_detection/include/CGAL/Polygon_mesh_processing/region_growing.h

```diff
@@ -0,0 +1,338 @@
+      \cgalParamDefault{1}
+    \cgalParamNEnd
+    \cgalParamNBegin{maximum_angle}
+      \cgalParamDescription{the maximum angle in degrees between two adjacent segments}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{25 degrees}
+    \cgalParamNEnd
+    \cgalParamNBegin{cosine_value}
+      \cgalParamDescription{the cos value computed as `cos(maximum_angle * PI / 180)`,
+      this parameter can be used instead of the `maximum_angle`}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{`cos(25 * PI / 180)`}
+    \cgalParamNEnd
+  \cgalNamedParamsEnd
+ */
+template <class PolygonMesh,
+          class RegionMap,
+          class CornerIdMap,
+          typename CGAL_NP_TEMPLATE_PARAMETERS>
+std::size_t
+detect_corners_of_regions(
+  const PolygonMesh& mesh,
+  RegionMap region_map,
+  std::size_t nb_regions,
+  CornerIdMap corner_id_map,
+  const CGAL_NP_CLASS& np = parameters::default_values())
+{
+  using parameters::choose_parameter;
+  using parameters::get_parameter;
+  using parameters::is_default_parameter;
+
+  typedef typename GetGeomTraits<PolygonMesh, CGAL_NP_CLASS>::type  Traits;
+
+  typedef boost::graph_traits<PolygonMesh> Graph_traits;
+  typedef typename Graph_traits::halfedge_descriptor halfedge_descriptor;
+  typedef typename Graph_traits::edge_descriptor edge_descriptor;
+  typedef typename Graph_traits::face_descriptor face_descriptor;
+  typedef typename Graph_traits::vertex_descriptor vertex_descriptor;
+
+  typedef typename boost::template property_map<PolygonMesh, CGAL::dynamic_edge_property_t<bool> >::type Default_ecm;
+  typedef typename internal_np::Lookup_named_param_def <
+    internal_np::edge_is_constrained_t,
+    CGAL_NP_CLASS,
+    Default_ecm
+  > ::type Ecm;
+
+  Default_ecm dynamic_ecm;
+  if(!(is_default_parameter<CGAL_NP_CLASS, internal_np::edge_is_constrained_t>::value))
+    dynamic_ecm = get(CGAL::dynamic_edge_property_t<bool>(), mesh);
+  Ecm ecm = choose_parameter(get_parameter(np, internal_np::edge_is_constrained), dynamic_ecm);
+
+  using Polyline_graph     = CGAL::Shape_detection::Polygon_mesh::Polyline_graph<PolygonMesh>;
+  using Segment_map        = typename Polyline_graph::Segment_map;
+  using Item               = typename Polyline_graph::Item;
+
+  using Line_region  = CGAL::Shape_detection::Segment_set::Least_squares_line_fit_region<Traits, Item, Segment_map>;
+  using Line_sorting = CGAL::Shape_detection::Segment_set::Least_squares_line_fit_sorting<Traits, Item, Polyline_graph, Segment_map>;
+  using RG_lines     = CGAL::Shape_detection::Region_growing<Polyline_graph, Line_region>;
+
+  // mark as constrained edges at the interface of two regions
+  for (edge_descriptor e : edges(mesh))
+  {
+    halfedge_descriptor h = halfedge(e, mesh);
+    face_descriptor f1 = face(h, mesh);
+    face_descriptor f2 = face(opposite(h, mesh), mesh);
+    if (f1 == Graph_traits::null_face() || f2 == Graph_traits::null_face() || get(region_map,f1)!=get(region_map,f2))
```

Comment:
`is_border_edge(h, mesh)` for clarity instead of `f1 == Graph_traits::null_face() || f2 == Graph_traits::null_face()`?

Reply:
I added a comment